config/configgrpc/configgrpc.go

```diff
@@ -436,14 +456,28 @@ func WithGrpcServerOption(opt grpc.ServerOption) ToServerOption {
 }
 func (grpcServerOptionWrapper) isToServerOption() {}
 
+type serverExtensionsOption struct {
+	extensions map[component.ID]component.Component
+}
+
+// WithServerExtensions is a [ToServerOption] which supplies the map of extensions to search for middlewares.
+//
+// Typically called with the output of host.GetExtensions().
+func WithServerExtensions(extensions map[component.ID]component.Component) ToServerOption {
+	return serverExtensionsOption{extensions: extensions}
+}
+func (serverExtensionsOption) isToServerOption() {}
```

Comment:
What do you think about combining WithClientExtensions and WithServerExtensions, so it's just WithExtensions returning a type that implements both interfaces? The names feel a bit awkward, since there's no such thing as client and server extensions

Reply:
The latter is what I had in mind - I was thinking `ToCommonOption` FWIW